File: pdb_fce.py
import requests
from bs4 import BeautifulSoup

compound = "aspirin"
url = f"https://www.ncbi.nlm.nih.gov/pccompound?term={compound}"

response = requests.get(url)

if response.status_code == 200:
    html_content = response.text
    soup = BeautifulSoup(html_content, 'html.parser')

    # Find all elements with class 'rprt'
    rprt_elements = soup.find_all('div', class_='rprt')

    # Get the first element (top entry)
    rprt = rprt_elements[0]

    # Find the 'rprtid' element
    rprtid_element = rprt.find('dl', class_='rprtid')
    cid = rprtid_element.find('dd').text
    print(f"CID: {cid}")
else:
    print("Error fetching the search results. Status code:", response.status_code)

File: pdb_fce.js
import fetch from 'node-fetch';
import { JSDOM } from 'jsdom';

const compound = "aspirin";
const url = `https://www.ncbi.nlm.nih.gov/pccompound?term=${compound}`;

fetch(url)
  .then(response => {
    if (response.ok) {
      return response.text();
    } else {
      throw new Error(`Error fetching the search results. Status code: ${response.status}`);
    }
  })
  .then(html => {
    const { window } = new JSDOM(html);
    const { document } = window;

    // Find all elements with class 'rprt'
    const rprtElements = document.querySelectorAll('.rprt');

    // Get the first element (top entry)
    const rprt = rprtElements[0];

    // Find the 'rprtid' element
    const rprtidElement = rprt.querySelector('.rprtid');
    const cid = rprtidElement.querySelector('dd').textContent;
  })
  .catch(error => console.error(error));
